pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolHandler.java

```diff
@@ -0,0 +1,340 @@
+    private ScheduledFuture<?> deadlineFuture = new NoopScheduledFuture();
+    /** Whether the next incoming message is compressed. */
+    private boolean isCompressed;
+    /** The encoding as determined by the grpc-encoding header. Will not be null. */
+    private Encoding encoding;
+    /** The current index into {@link #entityBytes} into which data is to be read. */
+    private int entityBytesIndex = 0;
+    /**
+     * The bytes of the next incoming message. This is created dynamically as a message is received, and is never
+     * larger than the system configured {@link PbjConfigBlueprint#maxMessageSize()}.
+     */
+    private byte[] entityBytes = null;
+    private BlockingQueue<Bytes> incomingMessages;
+
+    /** Create a new instance */
+    PbjProtocolHandler(final @NonNull HttpPrologue prologue,
+                       final @NonNull Http2Headers headers,
+                       final @NonNull Http2StreamWriter streamWriter,
+                       final int streamId,
+                       final @NonNull Http2Settings serverSettings,
+                       final @NonNull Http2Settings clientSettings,
+                       final @NonNull StreamFlowControl flowControl,
+                       final @NonNull Http2StreamState currentStreamState,
+                       final @NonNull PbjMethodRoute route,
+                       final @NonNull DeadlineDetector deadlineDetector) {
+
+        this.prologue = requireNonNull(prologue);
+        this.headers = requireNonNull(headers);
+        this.streamWriter = requireNonNull(streamWriter);
+        this.streamId = streamId;
+        this.serverSettings = requireNonNull(serverSettings);
+        this.clientSettings = requireNonNull(clientSettings);
+        this.flowControl = requireNonNull(flowControl);
+        this.currentStreamState = requireNonNull(currentStreamState);
+        this.route = requireNonNull(route);
+        this.deadlineDetector = requireNonNull(deadlineDetector);
+    }
+
+    @Override
+    public void init() {
+        try {
+            // If the grpc-timeout header is present, determine when that timeout would occur.
+            final var timeout = headers.httpHeaders().value(GRPC_TIMEOUT);
+            if (timeout.isPresent()) {
+                final var matcher = GRPC_TIMEOUT_PATTERN.matcher(timeout.get());
+                if (matcher.matches()) {
+                    final var num = Integer.parseInt(matcher.group(0));
+                    final var unit = matcher.group(1);
+                    final var deadline = System.nanoTime() + num * switch (unit) {
+                        case "H" -> 3600_000_000_000L;
+                        case "M" -> 60_000_000_000L;
+                        case "S" -> 1_000_000_000L;
+                        case "m" -> 1_000_000L;
+                        case "u" -> 1_000L;
+                        case "n" -> 1L;
+                        default -> throw new IllegalArgumentException("Invalid unit: " + unit);
```

Comment:
Might want to use the `TimeUnit.convert()` instead of hard-coding a bunch of magic numbers.